.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,16 @@
+name: Formatting and Setup.py Checks
+
+on:
+  pull_request:
+  push:
+    branches: [master, feature/package-refactoring]
```

Comment:
You can, but it's not recommended, and this is the expected behaviour of tests, if you look at the [starter template for python workflows](https://github.com/actions/starter-workflows/blob/master/ci/python-package.yml) they only run on a push/PR to master/the default branch. And if you look at [other packages using workflows for their python tests](https://github.com/search?q=uses%3A+actions%2Fsetup-python&type=code) basically all of them only run on push to master or on a PR.

One reason for this is that it's somewhat bad etiquette to run tests on push for every single branch when using the public runners, since (typically) the only time you're interested in the test results is when you're doing a PR or when you're merging to master, and triggering the CI for a push to anything uses up a lot of resources.

Another reason is that this can/will break stuff.For example if we end up having a `gh-pages` branch for hosting documentation, you wouldn't want to run the tests each time that changes. Also, if you have a workflow that pushes changes (like documentation builds) then you'll end up in an infinite loop of pushes triggering a PR that triggers a push.